community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/commands/expressions/ToIntFunction.scala

```diff
@@ -37,16 +37,12 @@ case class ToIntFunction(a: Expression) extends NullInNullOutExpression(a) {
       v.longValue()
     case v: String =>
       try {
-        v.toLong
+        val d = BigDecimal(v)
```

Comment:
But that was the problem, we had no way of knowing if the NumberFormatException was due to overflow or if was a float, _i.e._ `"1.2".toLong` or `"10508455564958384115".toLong. If it was the former it is ok to do `"1.2".toDouble.toLong` but not on the latter since `toDouble` is a lossy operation. For example
```
 scala> Long.MaxValue
res4: Long = 9223372036854775807

scala> "9223372036854775808".toDouble.toLong
res5: Long = 9223372036854775807
```